dev/publish_release.sh

```diff
@@ -0,0 +1,149 @@
+# http://www.apache.org/dev/release-signing.html
+# http://www.apache.org/dev/publishing-maven-artifacts.html
+
+if [[ -z "${SHASUM}" ]]; then
+  SHASUM="/usr/bin/shasum -a 1"
+fi
+
+if [[ -z "${WORKING_DIR}" ]]; then
+  WORKING_DIR=/tmp/zeppelin-release
+fi
+
+if [ $# -ne 2 ]; then
+  echo "usage) $0 [Release version] [Tag]"
+  echo "   ex. $0 0.6.0 v0.6.0"
+  exit 1
+fi
+
+for var in GPG_PASSPHRASE ASF_USERID ASF_PASSWORD
+do
+  if [[ -z "${!var}" ]]; then
+    echo "You need ${var} variable set"
+    exit 1
+  fi
+done
+
+export MAVEN_OPTS="-Xmx2g -XX:MaxPermSize=512m"
+
+RELEASE_VERSION="${1}"
+GIT_TAG="${2}"
+
+PUBLISH_PROFILES="-Pspark-1.6 -Phadoop-2.4 -Pyarn -Ppyspark -Psparkr -Pr"
+PROJECT_OPTIONS="-pl !zeppelin-server,!zeppelin-distribution"
+NEXUS_STAGING="https://repository.apache.org/service/local/staging"
+NEXUS_PROFILE="153446d1ac37c4"
+
+if [[ -d "${WORKING_DIR}" ]]; then
+  echo "Dir ${WORKING_DIR} already exists"
+  exit 1
+fi
+
+mkdir ${WORKING_DIR}
+
+echo "Cloning the source"
+# clone source
+git clone https://git-wip-us.apache.org/repos/asf/zeppelin.git ${WORKING_DIR}/zeppelin
+
+if [[ $? -ne 0 ]]; then
+  echo "Can not clone source repository"
+  exit 1
+fi
+
+cd ${WORKING_DIR}/zeppelin
+git checkout ${GIT_TAG}
+echo "Checked out ${GIT_TAG}"
+
+# remove unnecessary files
+rm ${WORKING_DIR}/zeppelin/.gitignore
+rm -rf ${WORKING_DIR}/zeppelin/.git
+
+function publish_to_maven() {
+  cd ${WORKING_DIR}/zeppelin
+
+  # Force release version
+  mvn versions:set -DnewVersion=$RELEASE_VERSION
+
+  # Using Nexus API documented here:
+  # https://support.sonatype.com/hc/en-us/articles/213465868-Uploading-to-a-Staging-Repository-via-REST-API
+  echo "Creating Nexus staging repository"
+  repo_request="<promoteRequest><data><description>Apache Zeppelin $RELEASE_VERSION</description></data></promoteRequest>"
+  out=$(curl -X POST -d "$repo_request" -u $ASF_USERID:$ASF_PASSWORD \
+    -H "Content-Type:application/xml" -v \
+    $NEXUS_STAGING/profiles/$NEXUS_PROFILE/start)
+  staged_repo_id=$(echo $out | sed -e "s/.*\(orgapachezeppelin-[0-9]\{4\}\).*/\1/")
+  echo "Created Nexus staging repository: $staged_repo_id"
+
+  tmp_repo=$(mktemp -d /tmp/zeppelin-repo-XXXXX)
+
+  echo "mvn clean install -Ppublish-distr \
+    -Dmaven.repo.local=$tmp_repo \
+    $PUBLISH_PROFILES $PROJECT_OPTIONS"
+  mvn clean install -Ppublish-distr -Dmaven.repo.local=$tmp_repo \
+    $PUBLISH_PROFILES $PROJECT_OPTIONS
+  if [[ $? -ne 0 ]]; then
+    echo "Build failed."
+    exit 1
+  fi
+
+  pushd $tmp_repo/org/apache/zeppelin
+  find . -type f | grep -v \.jar$ | grep -v \.pom$ |grep -v \.war$ | xargs rm
+
+  echo "Creating hash and signature files"
+  for file in $(find . -type f)
+  do
+    echo $GPG_PASSPHRASE | gpg --passphrase-fd 0 --output $file.asc \
+      --detach-sig --armor $file;
+    md5 -q $file > $file.md5
+    $SHASUM $file | cut -f1 -d' ' > $file.sha1
+  done
+
+  nexus_upload=$NEXUS_STAGING/deployByRepositoryId/$staged_repo_id
```

Comment:
We might want to follow the [conventions from our styleguide](https://github.com/apache/zeppelin/blob/master/CONTRIBUTING.md#code-convention) in particular on [Quoting](https://google.github.io/styleguide/shell.xml#Quoting) here